model.py

```diff
@@ -209,12 +209,9 @@ def forward(
             else:
                 scores = softmax_1(scores.float()).type_as(xq)
 
-            if FLAGS['inspect_softmax_sum']:
-                global softmax_sum
-                if len(softmax_sum) >=12: # num_blocks hardcoded
-                    softmax_sum = []
-                sums = scores.detach().squeeze(0).sum(-1).cpu()
-                softmax_sum.append(sums)
+            # save intermediate softmax tensor for compute_softmax_metrics during eval state
+            if not self.training:
+                self.softmax_output = scores.detach().cpu()
```

Comment:
Can we define `softmax_output` at init?

Reply:
You're probably right. I referred to https://github.com/tcapelle/llama2.c/blob/af8597ca6ac0c74142951c44b8d6926e170d791a/model.py when implementing the compute metrics and the self.output was not initialized under init, which in hindsight is probably not that good practice. I'll make another commit to define them at init.